sdks/java/io/solace/src/main/java/org/apache/beam/sdk/io/solace/broker/BasicAuthJcsmpSessionService.java

```diff
@@ -0,0 +1,148 @@
+ *
+ * <p>This class provides a way to connect to a Solace broker and receive messages from a queue. The
+ * connection is established using basic authentication.
+ */
+public class BasicAuthJcsmpSessionService implements SessionService {
+  private final String queueName;
+  private final String host;
+  private final String username;
+  private final String password;
+  private final String vpnName;
+  @Nullable private JCSMPSession jcsmpSession;
+  private final RetryCallableManager retryCallableManager = RetryCallableManager.create();
+
+  /**
+   * Creates a new {@link BasicAuthJcsmpSessionService} with the given parameters.
+   *
+   * @param queueName The name of the queue to receive messages from.
+   * @param host The host name or IP address of the Solace broker. Format: Host[:Port]
+   * @param username The username to use for authentication.
+   * @param password The password to use for authentication.
+   * @param vpnName The name of the VPN to connect to.
+   */
+  public BasicAuthJcsmpSessionService(
+      String queueName, String host, String username, String password, String vpnName) {
+    this.queueName = queueName;
+    this.host = host;
+    this.username = username;
+    this.password = password;
+    this.vpnName = vpnName;
+  }
+
+  @Override
+  public void connect() {
+    retryCallableManager.retryCallable(this::connectSession, ImmutableSet.of(JCSMPException.class));
+  }
+
+  @Override
+  public void close() {
+    if (jcsmpSession != null && !jcsmpSession.isClosed()) {
+      retryCallableManager.retryCallable(
+          () -> {
+            if (jcsmpSession != null) {
+              jcsmpSession.closeSession();
+            }
+            return 0;
+          },
+          ImmutableSet.of(IOException.class));
+    }
+  }
+
+  @Override
+  public MessageReceiver createReceiver() {
+    return retryCallableManager.retryCallable(
+        this::createFlowReceiver, ImmutableSet.of(JCSMPException.class));
+  }
+
+  @Override
+  public boolean isClosed() {
+    return jcsmpSession == null || jcsmpSession.isClosed();
+  }
+
+  private MessageReceiver createFlowReceiver() throws JCSMPException, IOException {
+    if (isClosed()) {
+      connectSession();
+    }
+
+    Queue queue = JCSMPFactory.onlyInstance().createQueue(queueName);
+
+    ConsumerFlowProperties flowProperties = new ConsumerFlowProperties();
+    flowProperties.setEndpoint(queue);
+    flowProperties.setAckMode(JCSMPProperties.SUPPORTED_MESSAGE_ACK_CLIENT);
+
+    EndpointProperties endpointProperties = new EndpointProperties();
+    endpointProperties.setAccessType(EndpointProperties.ACCESSTYPE_NONEXCLUSIVE);
+    if (jcsmpSession != null) {
+      return new SolaceMessageReceiver(
+          createFlowReceiver(jcsmpSession, flowProperties, endpointProperties));
+    } else {
+      throw new IOException(
+          "SolaceIO.Read: Could not create a receiver from the Jcsmp session: session object is null.");
+    }
```

Comment:
Since this method returns as a result of satisfying `if`, `else` is not needed and could just have the throw as the last line of the method.

Reply:
Agreed.